tests/e2e/support/cursor.rs

```diff
@@ -43,8 +52,29 @@ pub fn cursors(text: &str) -> (String, Cursors) {
         }
     }
 
+    assert_eq!(selections.len() % 2, 0);
+
+    assert!(matches!(selections.first(), Some(SelectionStackElement::Open(_)) | None));
+
+    selections.into_iter().tuple_windows().for_each(|window| match window {
+        (SelectionStackElement::Open(_), SelectionStackElement::Open(_))
+        | (SelectionStackElement::Close(_), SelectionStackElement::Close(_)) => {
```

Comment:
if you don't support nested selections then you don't need to accumulate a stack of events - you effectively support either one open selection, that you can immediately close, or empty state